File: contests/booking_codesprint/mobile/DestinationTogether3.java
import java.io.*;
import java.util.*;
import java.text.*;
import java.math.*;
import java.util.regex.*;

//
// https://www.hackerrank.com/contests/booking-com-passions-hacked-mobile/challenges/destination-together-3-1
//

public class Solution1 {

    public static long Fact(int n)
    {
        if (n == 0) return 1;
        return Fact(n - 1) * n;
    }
    
    public static void main(String[] args) throws IOException {
        BufferedReader in = new BufferedReader(new InputStreamReader(System.in));
        String s = in.readLine();
        String[] ss = s.split("\\s+");
        
        int N = Integer.parseInt(ss[0]) - 1;
        int M = Integer.parseInt(ss[1]) - 1;
        int C = Integer.parseInt(ss[2]) - 1;

        System.out.println(Fact(N - C + M - C + C));
    }
}

File: contests/booking_codesprint/backend/DestinationTogether3.cs
using System;
using System.Collections.Generic;
using System.IO;
using System.Linq;
/// <summary>
/// https://www.hackerrank.com/contests/booking-passions-hacked-backend/challenges/destinations-together-3
/// </summary>
class Solution1
{
    static long Fact(int n)
    {
        if (n == 0) return 1;
        return Fact(n - 1) * n;
    }
    static void Main(String[] args)
    {
        TextReader tIn = Console.In;
        TextWriter tOut = Console.Out;

        tIn = new StringReader(@"3 4 2");

        int[] nmc = tIn.ReadLine().Split().Select(p => int.Parse(p)).ToArray();
        int N = nmc[0] - 1;
        int M = nmc[1] - 1;
        int C = nmc[2] - 1;

        tOut.WriteLine(Fact(N - C + M - C + C));

        tIn.ReadLine();
    }
}
